pdfgen.c

```diff
@@ -2403,3 +2477,264 @@ int pdf_add_rgb24(struct pdf_doc *pdf, struct pdf_object *page, float x,
 
     return pdf_add_image(pdf, page, obj, x, y, display_width, display_height);
 }
+
+int pdf_add_png(struct pdf_doc *pdf, struct pdf_object *page,
+                 int x, int y, int display_width, int display_height,
+                 const char *png_file)
+{
+    const uint8_t png_signature[] = {0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A};
+    const char png_chunk_header[] =   "IHDR";
+    const char png_chunk_data[] =     "IDAT";
+    const char png_chunk_end[] =      "IEND";
+    //const char png_chunk_palette[] =  "PLTE";
+    struct pdf_object *obj;
+    uint8_t *png_data;
+    uint8_t *final_data;
+    int written = 0;
+    uint32_t pos;
+    struct png_chunk *chunk;
```

Comment:
Can you move the declaration of chunk inside the while loop? cppcheck is complaining about it.